src/common/recordbatch/src/adapter.rs

```diff
@@ -160,6 +160,7 @@ enum AsyncRecordBatchStreamAdapterState {
     Failed,
 }
 
+// TODO(ruihang): figure out what this adapter adapts to.
```

Comment:
I remember it's because we have to change `execute` method in `PhysicalPlan` from async to sync? Seems Datafusion requested it.